aptos-move/framework/supra-framework/sources/multisig_voting.move

```diff
@@ -0,0 +1,1219 @@
+/// 5. A voter, through the governance module, can call Voting::vote on a proposal. vote requires passing a &ProposalType
+/// and thus only the governance module that registers ProposalType can call vote.
+/// 6. Once the proposal's expiration time has passed and more than the defined threshold has voted yes on the proposal,
+/// anyone can call resolve which returns the content of the proposal (of type ProposalType) that can be used to execute.
+/// 7. Only the resolution script with the same script hash specified in the proposal can call Voting::resolve as part of
+/// the resolution process.
+module supra_framework::multisig_voting {
+    use std::bcs::to_bytes;
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::string::{String, utf8};
+    use std::vector;
+
+    use aptos_std::from_bcs::to_u64;
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_std::table::{Self, Table};
+    use aptos_std::type_info::{Self, TypeInfo};
+
+    use supra_framework::account;
+    use supra_framework::event::{Self, EventHandle};
+    use supra_framework::timestamp;
+    use supra_framework::transaction_context;
+    use aptos_std::from_bcs;
+
+    /// Current script's execution hash does not match the specified proposal's
+    const EPROPOSAL_EXECUTION_HASH_NOT_MATCHING: u64 = 1;
+    /// Proposal cannot be resolved. Either voting duration has not passed, not enough votes, or fewer yes than no votes
+    const EPROPOSAL_CANNOT_BE_RESOLVED: u64 = 2;
+    /// Proposal cannot be resolved more than once
+    const EPROPOSAL_ALREADY_RESOLVED: u64 = 3;
+    /// Proposal cannot contain an empty execution script hash
+    const EPROPOSAL_EMPTY_EXECUTION_HASH: u64 = 4;
+    /// Proposal's voting period has already ended.
+    const EPROPOSAL_VOTING_ALREADY_ENDED: u64 = 5;
+    /// Voting forum has already been registered.
+    const EVOTING_FORUM_ALREADY_REGISTERED: u64 = 6;
+    /// Resolution of a proposal cannot happen atomically in the same transaction as the last vote.
+    const ERESOLUTION_CANNOT_BE_ATOMIC: u64 = 8;
+    /// Cannot vote if the specified multi-step proposal is in execution.
+    const EMULTI_STEP_PROPOSAL_IN_EXECUTION: u64 = 9;
+    /// If a proposal is multi-step, we need to use `resolve_proposal_v2()` to resolve it.
+    /// If we use `resolve()` to resolve a multi-step proposal, it will fail with EMULTI_STEP_PROPOSAL_CANNOT_USE_SINGLE_STEP_RESOLVE_FUNCTION.
+    const EMULTI_STEP_PROPOSAL_CANNOT_USE_SINGLE_STEP_RESOLVE_FUNCTION: u64 = 10;
+    /// If we call `resolve_proposal_v2()` to resolve a single-step proposal, the `next_execution_hash` parameter should be an empty vector.
+    const ESINGLE_STEP_PROPOSAL_CANNOT_HAVE_NEXT_EXECUTION_HASH: u64 = 11;
+    /// Cannot call `is_multi_step_proposal_in_execution()` on single-step proposals.
+    const EPROPOSAL_IS_SINGLE_STEP: u64 = 12;
+
+    /// ProposalStateEnum representing proposal state.
+    const PROPOSAL_STATE_PENDING: u64 = 0;
```

Comment:
```suggestion
    const EPROPOSAL_STATE_PENDING: u64 = 0;
```
always use `E` as a prefix to any constant which is supposed to represent an error code.

Reply:
`PROPOSAL_STATE_PENDING`, `PROPOSAL_STATE_SUCCEEDED`, and `PROPOSAL_STATE_FAILED` are not actually errors; they simply represent the states of a proposal—whether it's pending, successful, or failed.
Therefore, adding an 'E' prefix to these variable names may not be appropriate.